newrelic/newrelic-dotnet-agent
# Fix azure site extension

### Description
This PR is the fix for the issue [#399](https://github.com/newrelic/newrelic-dotnet-agent/issues/399). It does the following:
1. Removes "NewRelicCorePackage" and "NewRelicPackage" directories and its contents post installation.
2. Removes `libgrpc_csharp_ext.x64.so` and `libNewRelicProfiler.so` as they won't be used. Currently, only apps that hosted in Windows environment can install Site Extensions. Moreover, our site extension install script is a powershell script, it does not run on Linux.
3. Add GHA release/deploy for the New Relic Azure Site Extension.

By removing unused artifacts and files, it saves ~400MB space.
### Testing
Manually tested. Testing steps:
1. Build the New Relic Azure Site Extension as a nuget package with updated install script.
2. Upload the site extension nuget package to a private nuget manager site (I used MyGet.org).
3. Create a test Asp.Net Core website and a test Aspt.Net website in Azure.
4. Add an app setting called SCM_SITEEXTENSIONS_FEED_URL, and set its value to your MyGet feed.  Then click “save”.
5. Add app settings called NEW_RELIC_LICENSE_KEY and NEW_RELIC_HOST, NEW_RELIC_APP_NAME with appropriate value.
6. Install the new site extension.
7. Validate if the test apps report to New Relic.
8. Login to Kudu, check if .\SiteExtensions\NewRelic.Azure.WebSites.Extension no longer contains "NewRelicCorePackage" and "NewRelicPackage" directories.
9. In Kudu, check for where the Core Agent installed no longer contains linux bits (.so files.)

### Notes:
Since this PR removes the linux grpc lib as it won't be used, I still tested 8T to make sure it doesn't break anything. I found 8T wasn't working for my test app. The agent had problem connecting the NR trace observer and logged out the following messages.
```
2021-03-03 22:49:52,583 NewRelic  DEBUG: [pid: 8852, tid: 5] SpanStreamingService: Timeout creating gRPC channel at endpoint xxxxx-xxx-xxx-xxx.aws-us-east-2.tracing.staging-edge.nr-data.net:443. (attempt 0, timeout 10000ms) 
2021-03-03 22:49:52,587 NewRelic FINEST: [pid: 8852, tid: 5] SpanStreamingService: Backoff for 15000ms before attempting to reconnect. 
```
However, this happened with/without the linux grpc (I tested with the old site extension and still got the same error). This could be from my test app set up or Azure specific issue with grpc.

Created an [issue](https://github.com/newrelic/newrelic-dotnet-agent/issues/479) to follow up.
